src/bsoncxx/CMakeLists.txt

```diff
@@ -138,11 +138,10 @@ if(BSONCXX_BUILD_SHARED)
 
     if(WIN32)
         # Add resource-definition script for Windows shared library (.dll).
-        set_property(TARGET bsoncxx_shared APPEND PROPERTY SOURCES bsoncxx.rc)
-        set(bsoncxx_rc_definitions "")
-        list(APPEND bsoncxx_rc_definitions BSONCXX_VERSION_STR="${BSONCXX_VERSION}")
-        list(APPEND bsoncxx_rc_definitions BSONCXX_OUTPUT_BASENAME="${BSONCXX_OUTPUT_BASENAME}")
-        set_source_files_properties(bsoncxx.rc PROPERTIES COMPILE_DEFINITIONS "${bsoncxx_rc_definitions}")
+        set(BSONCXX_VERSION_STR "${BSONCXX_VERSION}")
+        set(BSONCXX_OUTPUT_BASENAME "${BSONCXX_OUTPUT_BASENAME}")
+        configure_file(bsoncxx.rc.in bsoncxx.rc)
+        set_property(TARGET bsoncxx_shared APPEND PROPERTY SOURCES ${CMAKE_CURRENT_BINARY_DIR}/bsoncxx.rc)
```

Comment:
Can this be `target_sources(bsoncxx_shared PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/bsoncxx.rc)` instead?